third_party/forked/golang/net/ip_test.go

```diff
@@ -2,26 +2,38 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !js
-// +build !js
-
 package net
 
+///////////////////////////////////////////////////////////////////////////////
+// NOTE: This file was forked because we need to maintain backwards-compatible
+// IP parsing logic, which was changed in a correct but incompatible way in
+// go-1.17.
+//
+// See https://issue.k8s.io/100895
+///////////////////////////////////////////////////////////////////////////////
+
 import (
-	"bytes"
-	"math/rand"
+	stdnet "net"
 	"reflect"
-	"runtime"
 	"testing"
 )
 
+//
+// Lean on the standard net lib as much as possible.
+//
+type IPMask = stdnet.IPMask
+
+var IPv4Mask = stdnet.IPv4Mask
+
 var parseIPTests = []struct {
 	in  string
 	out IP
 }{
 	{"127.0.1.2", IPv4(127, 0, 1, 2)},
 	{"127.0.0.1", IPv4(127, 0, 0, 1)},
+	{"127.001.002.003", IPv4(127, 1, 2, 3)}, // see https://issue.k8s.io/100895
```

Comment:
one that demonstrates it parses as decimal as well?

```
{"127.010.020.030", IPv4(127, 10, 20, 30)}, // see https://issue.k8s.io/100895
```

Reply:
good call.  Also will test 007 vs 008 to prove octal's edge